mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,263 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import ast
+import json
+import os
+import re
+from botocore.vendored import requests
+import logging
+import secret_manager
+
+
+class LabelBot:
+    LABEL_PAGE_PARSE = 30  # Limit for total labels per page to parse
+
+    def __init__(self,
+                 repo=os.environ.get("repo"),
+                 github_user=None,
+                 github_oauth_token=None,
+                 apply_secret=True):
+        """
+        Initializes the Label Bot
+        :param repo: GitHub repository that is being referenced
+        :param github_user: GitHub username
+        :param github_oauth_token: GitHub authentication token (Personal access token)
+        :param apply_secret: GitHub secret credential (Secret credential that is unique to a GitHub developer)
+        """
+        self.repo = repo
+        self.github_user = github_user
+        self.github_oauth_token = github_oauth_token
+        if apply_secret:
+            self._get_secret()
+        self.auth = (self.github_user, self.github_oauth_token)
+        self.all_labels = None
+
+    def _get_rate_limit(self):
+        """
+        This method gets the remaining rate limit that is left from the GitHub API
+        :return Remaining API requests left that GitHub will allow
+        """
+        res = requests.get('https://api.github.com/{}'.format('rate_limit'),
+                           auth=self.auth)
+        res.raise_for_status()
+        data = res.json()['rate']
+        return data['remaining']
+
+    def _get_secret(self):
+        """
+        This method is to get secret value from Secrets Manager
+        """
+        secret = json.loads(secret_manager.get_secret())
+        self.github_user = secret["github_user"]
+        self.github_oauth_token = secret["github_oauth_token"]
+
+    def _tokenize(self, string):
+        """
+        This method is to extract labels from comments
+        :param string: String parsed from a GitHub comment
+        :return Set of Labels which have been extracted
+        """
+        substring = string[string.find('[') + 1: string.rfind(']')]
+        labels = [' '.join(label.split()) for label in substring.split(',')]
+        return labels
+
+    def _clean_string(self, raw_string, sub_string):
```

Comment:
a better name would be "ascii_only"   clean can be many things (for different people :)